Enable usage of MotionMark 1.2 and Speedometer 3 in compare-results

jcarterbohan opened the pull request:
#### de4f30975494b850fa1bae1d84151d71e34a335c
<pre>
Enable usage of MotionMark 1.2 and Speedometer 3 in compare-results
<a href="https://bugs.webkit.org/show_bug.cgi?id=260311">https://bugs.webkit.org/show_bug.cgi?id=260311</a>

Reviewed by Dewei Zhu.

This adds support for comparing results for MotionMark 1.2 and Speedometer 3.
The implementation is identical to their previous versions because their results are formatted the same.
Minor refactoring around MotionMark results detection.

* Tools/Scripts/compare-results:
(readJSONFile):
(speedometer3Breakdown):
(speedometer3BreakdownSyncAsync):
(motionMarkBreakdown):
(JetStream3Results):
(detectSpeedometer2):
(detectSpeedometer3):
(Speedometer2Results):
(Speedometer3Results):
(detectMotionMark1_1_1):
(detectMotionMark1_2):
(motionMarkResults):
(detectBenchmark):
(biggerIsBetter):
(main):

Canonical link: <a href="https://commits.webkit.org/267291@main">https://commits.webkit.org/267291@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6f8baf9ed7024ed347f35f57716815ae3b389a40

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16202 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16520 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16933 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17966 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15205 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/16389 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/19585 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16632 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/17599 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16396 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16846 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13843 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18732 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14089 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14661 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/21497 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15078 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14826 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/18057 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15423 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/13079 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14650 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3870 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19015 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15245 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

xw commented on Tools/Scripts/compare-results:
```diff
@@ -466,14 +512,20 @@ def detectMotionMark1_1(payload):
+
 def motionMarkResults(payload):
-    assert detectMotionMark(payload) or detectMotionMark1_1(payload) or detectMotionMark1_1_1(payload)
+    assert detectMotionMark(payload) or detectMotionMark1_1(payload) or detectMotionMark1_1_1(payload) or detectMotionMark1_2(payload)
```
Nit: This can be rewritten as:
```
assert any(validMotionMarkDetector(payload) for validMotionMarkDetector in [detectMotionMark, detectMotionMark1_1, detectMotionMark1_1_1, detectMotionMark1_2])
```

dewei-zhu commented on Tools/Scripts/compare-results:
```diff
@@ -393,16 +427,28 @@ def JetStream3Results(payload):
     return results
 
+
+
```
Nit: remove this line so that only two lines between functions.
Hmm, this file seems to have inconsistent style, might worth a separate change to unify that.

jcarterbohan replied:
Just to clarify, do you want the style changes (assuming its mostly the spacing between multiple functions) to be in this PR or a separate bug report?

xw replied:
IMO, the style of this script is inconsistent enough with the rest of our Python scripts that I would suggest doing them as a separate patch. Considering that the minimum is that we:

 a. Change functions and variable names to use underscores (to align with PEP 8 more)
 b. Remove unreachable code and reduce redundant/repeated code

These by themselves would account for so many changed lines that if we were to do these as a fly-by fix the diff for the current patch would become quite confusing.

xw commented on Tools/Scripts/compare-results:
```diff
@@ -643,7 +703,18 @@ def main():
+
+        if args.csv:
+            writeCSV(speedometer3Breakdown(a), speedometer3Breakdown(b), args.csv)
+    elif typeA == MotionMark or typeA == MotionMark1_1 or typeA == MotionMark1_1_1 or typeA == MotionMark1_2:
```
Nit: this can be rewritten as:
```
elif any(typeA == validMotionMarkTest for validMotionMarkTest in [MotionMark, MotionMark1_1, MotionMark1_1_1, MotionMark1_2]):
```

xw commented on Tools/Scripts/compare-results:
```diff
@@ -504,6 +558,8 @@ def detectBenchmark(payload):
+        return MotionMark1_2
     return None
 
 def biggerIsBetter(benchmarkType):
```
Nit: This can be written way more efficiently without a giant `if` block (also `MotionMark1_1_1` is missing from the original):
```
def biggerIsBetter(benchmarkType):
    if benchmarkType in [JetStream2, JetStream3, Speedometer2, Speedometer3, MotionMark, MotionMark1_1, MotionMark1_1_1, MotionMark1_2]:
        return True
    elif benchmarkType in [PLT5, CompetitivePLT, PLUM3]:
        return False
    else:
        raise Exception('An unknown benchmark type was passed into biggerIsBetter: {}. It should not be possible to hit this.'.format(benchmarkType))
```

xw reviewed (commented):
Welcome to the WebKit project, congrats on your first PR! (or at least, I assume that's what the "First-time contributor" badge means)

See my comment above - this patch is good, but given that we now have four MotionMark versions to keep a track of we could do with a bit of consolidation.

Other than these nitpicks, this patch looks good! Though worth noting that I'm not officially a Reviewer, you'll need to wait for someone with Reviewer status (like @dewei-zhu) to r+ your patch before requesting to be landed.

dewei-zhu reviewed (approved):
r=me

jcarterbohan committed:
Enable usage of MotionMark 1.2 and Speedometer 3 in compare-results
https://bugs.webkit.org/show_bug.cgi?id=260311

Reviewed by Dewei Zhu.

This adds support for comparing results for MotionMark 1.2 and Speedometer 3.
The implementation is identical to their previous versions because their results are formatted the same.
Minor refactoring around MotionMark results detection.

* Tools/Scripts/compare-results:
(readJSONFile):
(speedometer3Breakdown):
(speedometer3BreakdownSyncAsync):
(motionMarkBreakdown):
(JetStream3Results):
(detectSpeedometer2):
(detectSpeedometer3):
(Speedometer2Results):
(Speedometer3Results):
(detectMotionMark1_1_1):
(detectMotionMark1_2):
(motionMarkResults):
(detectBenchmark):
(biggerIsBetter):
(main):

Canonical link: https://commits.webkit.org/267291@main
